# **Detecting Phishing URL's**


Phishing is among the most concerning issues in a constantly changing world. The increasing use of the Internet has led to a new way of stealing data, known as cybercrime. Cybercrime refers to stealing private information and violating privacy through computers. The primary technique used is phishing. Phishing via URLs (Uniform Resource Locators) is one of the most common types, and its primary goal is to steal the data from the user when the user accesses the malicious website. This work aims to provide a solution for detecting such websites with the help of machine learning algorithms focused on the behaviors and qualities of the suggested URL.

Data Cleaning
- To ensure the data is structured and prepared for analysis

In [1]:
# import necessary libraries

import pandas as pd # for data manipulation and analysis
import numpy as np #  for working with arrays
import seaborn as sns # provides high-level interface for drawing attractive and informative statistical graphics
import matplotlib.pyplot as plt # creating static, animated and interactive visualizations in Python
%matplotlib inline

In [24]:
# load csv file

data = pd.read_csv('/phishing_site_urls.csv')

In [25]:
# check the first 5 rows of the data
data.head()

URL Label
0  nobell.it/70ffb52d079109dca5664cce6f317373782/...   bad
1  www.dghjdgf.com/paypal.co.uk/cycgi-bin/webscrc...   bad
2  serviciosbys.com/paypal.cgi.bin.get-into.herf....   bad
3  mail.printakid.com/www.online.americanexpress....   bad
4  thewhiskeydregs.com/wp-content/themes/widescre...   bad

In [26]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 549346 entries, 0 to 549345
Data columns (total 2 columns):
 #   Column  Non-Null Count   Dtype 
---  ------  --------------   ----- 
 0   URL     549346 non-null  object
 1   Label   549346 non-null  object
dtypes: object(2)
memory usage: 8.4+ MB


In [27]:
# check for any missing values
data.isnull().sum()

URL      0
Label    0
dtype: int64

In [28]:
data.describe()

URL   Label
count                          549346  549346
unique                         507195       2
top     jhomitevd2abj3fk.tor2web.org/    good
freq                               52  392924

Based on the output above, there are some observations to be made:
1.  There are a total of 549,346 rows of data
2.  The label column contains 2 unique ID

In [32]:
# identify the value counts in Label column
print(data.Label.value_counts())

good    392924
bad     156422
Name: Label, dtype: int64
